packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemContextMenu.tsx

```diff
@@ -172,8 +315,34 @@ export const TreeItemContextMenu = ({
             <ListItemText primary="Delete" />
           </MenuItem>
         ) : null}
+        {state.menuEntries.map((entry) => (
+          <MenuItem
+            key={entry.id}
+            onClick={(_) => invokeContextMenuEntry(entry)}
+            data-testid={`context-menu-entry-${entry.label}`}
+            disabled={readOnly}
+            aria-disabled>
+            <ListItemIcon>
+              {entry.iconURL.length > 0 ? (
+                <IconOverlay iconURL={entry.iconURL} alt={entry.label} title={entry.label} />
+              ) : (
+                <DeleteIcon style={{ opacity: 0 }} fontSize="small" />
```

Comment:
This icon has the opacity set to 0 so it could be whatever icon. 
This is the easiest way to ensure that even if no icon is specified, the label is at the right place and perfectly aligned with other menu entry labels.